AddressService/AddressService.Handlers/GetLocationsByDistanceHandler.cs

```diff
@@ -0,0 +1,73 @@
+﻿using AddressService.Core.Dto;
+using AutoMapper;
+using MediatR;
+using System.Threading;
+using System.Threading.Tasks;
+using AddressService.Core.Utils;
+using AddressService.Handlers.BusinessLogic;
+using HelpMyStreet.Contracts.AddressService.Request;
+using HelpMyStreet.Contracts.AddressService.Response;
+using HelpMyStreet.Utils.Utils;
+using AddressService.Core.Interfaces.Repositories;
+using System.Collections.Generic;
+using System;
+using System.Linq;
+using UserService.Core.Utils;
+
+namespace AddressService.Handlers
+{
+    public class GetLocationsByDistanceHandler : IRequestHandler<GetLocationsByDistanceRequest, GetLocationsByDistanceResponse>
+    {
+        private readonly IRepository _repository;
+        
+        public GetLocationsByDistanceHandler(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task<GetLocationsByDistanceResponse> Handle(GetLocationsByDistanceRequest request, CancellationToken cancellationToken)
+        {
+            var postcodeDetails = await _repository.GetPostcodeCoordinatesAsync(new List<string>() { request.Postcode });
+
+            if(postcodeDetails == null)
+            {
+                throw new Exception($"Unable to retrieve post code details for {request.Postcode}");
+            }
+
+            if(postcodeDetails.Count()!=1)
+            {
+                throw new Exception($"Only expecting 1 row in collection for {request.Postcode}. {postcodeDetails.Count()} rows returned");
+            }
+
+            var allLocations = _repository.GetAllLocations();
+
+            if (allLocations == null && allLocations.Count()==0)
+            {
+                throw new Exception($"Unable to retrieve any locations");
+            }
+
+            DistanceCalculator distanceCalculator = new DistanceCalculator();
+            PostcodeWithCoordinatesDto requestPostCode = postcodeDetails.First();
+
+            var locationDistances =  allLocations.Where(x => distanceCalculator.GetDistanceInMiles(
```

Comment:
I know you said GetDistanceInMiles is fast, but it would seem neater to me to calcualte the distances once, then keep the ones below the threshold.  I think the current implementation will calculate all distances once for filtering, then re-calculate distances for the ones we want to return.